tuf/ngclient/updater.py

```diff
@@ -1,7 +1,61 @@
 # Copyright 2020, New York University and the TUF contributors
 # SPDX-License-Identifier: MIT OR Apache-2.0
 
-"""TUF client workflow implementation.
+"""TUF module that implements the client update workflow.
+
+This module contains the Updater class that provides an implementation of the
+`TUF client workflow
+<https://theupdateframework.github.io/specification/latest/#detailed-client-workflow>`_.
+Updater provides an API to query available targets and to download them in a
+secure manner: All downloaded files are verified by signed metadata.
+
+High-level description of Updater functionality:
+  * Initializing an :class:`~tuf.ngclient.updater.Updater` loads and validates
+    the trusted local root metadata: This root metadata is used as the source
+    of trust for all other metadata.
+  * Calling :func:`~tuf.ngclient.updater.Updater.refresh()` will update root
+    metadata and load all other top-level metadata as described in the
+    specification, using both locally cached metadata and metadata downloaded
+    from the remote repository.
+  * When metadata is up-to-date, targets can be dowloaded. The repository
+    snapshot is consistent so multiple targets can be downloaded without
+    fear of repository content changing. For each target:
```

Comment:
I am not sure I understand that sentence: `The repository  snapshot is consistent so multiple targets can be downloaded without fear of repository content changing.`
What do you mean by consistent?

Reply:
that even if there is a longer period of time between downloading one targets metadata and another targets metadata, they will use metadata versions from the same snapshot: so there's no fear of accidentally downloading one file that was added into the repo after the update started. All target files that are downloaded are from one consistent snapshot.

Maybe this does not need to be here at all, not sure...

Reply:
Ahaaa, now it makes sense.
It's a good explanation for a new reader. I am used to that idea and have never thought about it, but it makes sense to capture it for new readers.
Maybe you can say it like that:
`The repository snapshot is consistent and can be reused so multiple targets can be downloaded without fear of repository content changing.`